File: src/IsbInterop.CSharp/ComponentTokens/Proxies/ComponentToken.cs
using IsbInterop.Base.Proxies;

namespace IsbInterop.ComponentTokens.Proxies
{
  /// <summary>
  /// Обертка над IComponentToken.
  /// </summary>
  internal class ComponentToken : EdmsObject<IComponentTokenInfo>, IComponentToken
  {
    /// <summary>
    /// Получает информацию об объекте.
    /// </summary>
    /// <returns>Информация об объекте.</returns>
    public override IComponentTokenInfo GetInfo()
    {
      var rcwIComponentTokenInfo = GetRcwObjectInfo();
      return new ComponentTokenInfo(rcwIComponentTokenInfo, Scope);
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwIComponentToken">СOM-объект IComponentToken.</param>
    /// <param name="scope">Область видимости.</param>
    public ComponentToken(object rcwIComponentToken, IScope scope)
      : base(rcwIComponentToken, scope) { }
  }
}

File: src/IsbInterop/ComponentTokens/Proxies/ComponentToken.vb
Imports IsbInterop.Base.Proxies

Namespace ComponentTokens.Proxies

  ''' <summary>
  ''' Обертка над IComponentToken.
  ''' </summary>
  Friend Class ComponentToken
    Inherits EdmsObject(Of IComponentTokenInfo)
    Implements IComponentToken

    ''' <summary>
    ''' Получает информацию об объекте.
    ''' </summary>
    ''' <returns>Информация об объекте.</returns>
    Public Overrides Function GetInfo() As IComponentTokenInfo
      Dim rcwIComponentTokenInfo = GetRcwObjectInfo()
      Return New ComponentTokenInfo(rcwIComponentTokenInfo, Scope)
    End Function

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwIComponentToken">СOM-объект IComponentToken.</param>
    ''' <param name="scope">Область видимости.</param>
    Public Sub New(rcwIComponentToken As Object, scope As IScope)
      MyBase.New(rcwIComponentToken, scope)
    End Sub
  End Class
End Namespace